### ISSUE TITLE
Automatic pdf build

### ISSUE THREAD

**Issue by dpritchLibre** [BarkleyBG/LCC#18]
We should have a way to automatically build the pdf.  This will be especially useful for people who just want to view a current snapshot.

What is the right way to do this? CMake? Something else?

**Comment by BarkleyBG** [comment]
I'm not sure I know the answer to that

**Comment by dpritchLibre** [comment]
According to my internet research, CMake is probably the best way to do construct a platform-independent build system in the general case.  Another thing to try to do is to use make and have it figure out what system you are usual and issue system-specific commands.

However both of these options seem far more difficult / complicated than it is worth.  Instead, I think we should consider the idea suggested in issue #5, namely that paste built versions of the pdf somewhere that people can access them through the internet, such as on http://www.bios.unc.edu/~bribark/.

We can probably use Travis to build the pdf's and update to the website whenever e.g. the master or develop branch is updated.

**Comment by BarkleyBG** [comment]
Any PDF should have

- [ ] title page #5 
- [ ] front matter including
    - [ ] contributor information #5 
    - [ ] date and version #21
    - [ ] information on how to contribute (i.e. link to the repo) #19 and #14
    - [ ] introduction to the document and its aims 
- [ ] ToC #6  (references and index can be added later so i'm not even adding a bullet point)

I'm thinking we should follow the git repo's model of master/next/pu branches. For us, maybe we could user master/next_update/distant_future?

- Any changes to next_update would need to get merged into distant_future. That should be easy enough.
- topic branches would be used to make changes to next_update and distant_future. Hotfix topic branches of course for master too.
- The one thing I don't know is how we would add a watermark to the PDF built from next_update that gets shared with readers for proofreading. 
    - Maybe we could have a next_update_mirror branch with the watermark on it? 
    - I worry that this would result in both the next_update and next_update_mirror branches would be changing in parallel. Perhaps we could simply merge often to keep things simple? 
    - Alternatively we could build the PDF in a commit in the next_update branch, mark it with a tag, and then delete the watermark in the next commit and continue forward? Hmmm...

**Comment by dpritchLibre** [comment]
To address one of the points in the above, if we use Travis then maybe we could ask it to add the watermark?

**Comment by BarkleyBG** [comment]
Closing  and migrating these open issues to #22
